src/Microsoft.ML.Data/Evaluators/EvaluatorUtils.cs

```diff
@@ -1353,15 +1353,43 @@ public static string GetConfusionTable(IHost host, IDataView confusionDataView,
             host.CheckValue(confusionDataView, nameof(confusionDataView));
             host.CheckParam(sample == -1 || sample >= 2, nameof(sample), "Should be -1 to indicate no sampling, or at least 2");
 
-            // Get the class names.
-            int countCol;
-            host.Check(confusionDataView.Schema.TryGetColumnIndex(MetricKinds.ColumnNames.Count, out countCol), "Did not find the count column");
-            var type = confusionDataView.Schema[countCol].Annotations.Schema.GetColumnOrNull(AnnotationUtils.Kinds.SlotNames)?.Type as VectorDataViewType;
-            host.Check(type != null && type.IsKnownSize && type.ItemType is TextDataViewType, "The Count column does not have a text vector metadata of kind SlotNames.");
+            var weightColumn = confusionDataView.Schema.GetColumnOrNull(MetricKinds.ColumnNames.Weight);
+            bool isWeighted = weightColumn.HasValue;
 
+            var confusionMatrix = GetConfusionTableAsType(host, confusionDataView, binary, sample, false);
+            var confusionTableString = GetConfusionTableAsString(confusionMatrix, false);
+
+            // if there is a Weight column, return the weighted confusionMatrix as well, from this function.
+            if (isWeighted)
+            {
+                confusionMatrix = GetConfusionTableAsType(host, confusionDataView, binary, sample, false);
+                weightedConfusionTable = GetConfusionTableAsString(confusionMatrix, true);
+            }
+            else
+                weightedConfusionTable = null;
+
+            return confusionTableString;
+        }
+
+        public static ConfusionMatrix GetConfusionTableAsType(IHost host, IDataView confusionDataView, bool binary = true, int sample = -1, bool getWeighted = false)
```

Comment:
>GetConfusionTableAsType [](start = 38, length = 23)

Maybe it make sense to rename this method to `GetConfusionMatrix` and `GetConfusionTable` to `GetConfusionMatrixAsString` or `GetConfusionString`?

Reply:
GetConfusionMatrixAsString exists. So   I used GetConfusionTableAsFormattedString. 

and I hate it already. lmk if anything better comes to mind. 

---
In reply to: [275080299](https://github.com/dotnet/machinelearning/pull/3250#discussion_r275080299) [](ancestors = 275080299)

Reply:
Can you at least rename this one to `GetConfusionMatrix` ?

---
In reply to: [275088372](https://github.com/dotnet/machinelearning/pull/3250#discussion_r275088372) [](ancestors = 275088372,275080299)